crates/cargo-contract/src/cmd/test.rs

```diff
@@ -35,16 +36,33 @@ pub struct TestCommand {
     /// Path to the `Cargo.toml` of the contract to test.
     #[clap(long, parse(from_os_str))]
     manifest_path: Option<PathBuf>,
+    /// Test all contract packages in the workspace.
+    #[clap(long = "--all")]
+    test_all: bool,
     #[clap(flatten)]
     verbosity: VerbosityFlags,
 }
 
 impl TestCommand {
-    pub fn exec(&self) -> Result<TestResult> {
+    pub fn exec(&self) -> Result<Vec<TestResult>> {
         let manifest_path = ManifestPath::try_from(self.manifest_path.as_ref())?;
         let verbosity = TryFrom::<&VerbosityFlags>::try_from(&self.verbosity)?;
 
-        execute(&manifest_path, verbosity)
+        let mut ret = Vec::new();
```

Comment:
We'll also want some sort of counter + name in the output of `cargo contract test`